server/routes/auth/sign_up.js

```diff
@@ -2,40 +2,32 @@ const express = require('express');
 const router = express.Router();
 const { User } = require('./../../models');
 const JWTService = require('./../../services/JWTService')
-const moment = require('moment')
+const Validator = require('./../../services/Validator');
 const handler = {
 
-  async sign_up(req,res) {
-    const {
-      nick,
-      first_name,
-      last_name,
-      email,
-      password,
-      avatar,
-      birth_date } = req.body;
-    const usersCount = await User.where({ email }).count();
-    if (usersCount) {
-      throw new Error('This email is already registered. How about to log in? ', 409);
-    }    
-    let userInfo = null;
-    let user = null;
-    userInfo = {
-      nick,
-      first_name,
-      last_name,
-      email,
-      password,
-      avatar,
-      birth_date,
-      created_at: moment().format('YYYY-MM-DD h:mm:ss')
-      };
-      user = await new User(userInfo).save(null);
-    res.status = 201;
-    res.send({
-      token: `Bearer ${JWTService.signUser(user)}`,
-      id: user.get('id')
+  async sign_up(req,res,next) {
+    const { email, password } = req.body;
+    const validation = await new Validator(req.body, {
+      email: 'required|email|unique',
+      password: 'required|min:3'
     });
+   
+    validation.fails(function() {
+      return next(validation.errors);
+    });
+    
+    try{
+      const user = await new User({email,password}).save(null);
```

Comment:
During the authorization, the email checks in lower case, should we store the email in the database, also in lower case?

Reply:
In the user model, when we get an email, we automatically convert to lowercase
file "server/models/User.js"  --20 lines

Reply:
Oy yes, I already see, line 17. Thank you)